sdk/client-core/http-jdk-httpclient/src/main/java/io/clientcore/http/jdk/httpclient/JdkHttpClient.java

```diff
@@ -137,28 +136,36 @@ private static <T> HttpResponse.BodyHandler<T> getResponseHandler(boolean hasRea
         return hasReadTimeout ? responseInfo -> timeoutSubscriber.apply(readTimeout.toMillis()) : jdkBodyHandler.get();
     }
 
-    private Response<?> toResponse(HttpRequest request, HttpResponse<InputStream> response,
-        ResponseBodyMode responseBodyMode) throws IOException {
+    private Response<?> toResponse(HttpRequest request, HttpResponse<InputStream> response) throws IOException {
         HttpHeaders coreHeaders = fromJdkHttpHeaders(response.headers());
 
         String contentType = coreHeaders.getValue(HttpHeaderName.CONTENT_TYPE);
-        if (ServerSentEventUtil.isTextEventStreamContentType(contentType)) {
+        if (ServerSentEventUtils.isTextEventStreamContentType(contentType)) {
             ServerSentEventListener listener = request.getServerSentEventListener();
 
             if (listener != null) {
-                processTextEventStream(request, this::send, response.body(), listener, LOGGER);
+                processTextEventStream(request, this, response.body(), listener, LOGGER);
             } else {
-                throw LOGGER.logThrowableAsError(new RuntimeException(ServerSentEventUtil.NO_LISTENER_ERROR_MESSAGE));
+                throw LOGGER.logThrowableAsError(new RuntimeException(ServerSentEventUtils.NO_LISTENER_ERROR_MESSAGE));
             }
 
-            return new JdkHttpResponse(request, response.statusCode(), coreHeaders, BinaryData.fromBytes(EMPTY_BODY));
+            return new JdkHttpResponse(request, response.statusCode(), coreHeaders, BinaryData.EMPTY);
         }
 
-        return processResponse(request, response, coreHeaders, contentType, responseBodyMode);
+        return processResponse(request, response, coreHeaders, contentType);
     }
 
     private Response<?> processResponse(HttpRequest request, HttpResponse<InputStream> response,
-        HttpHeaders coreHeaders, String contentType, ResponseBodyMode responseBodyMode) throws IOException {
+                                        HttpHeaders coreHeaders, String contentType) throws IOException {
+
+        RequestOptions requestOptions = request.getRequestOptions();
+
+        if (requestOptions == RequestOptions.NONE) {
+            requestOptions = new RequestOptions();
+        }
```

Comment:
This pattern exists because you need to mutate the response body mode - if we can do away with that, we will hopefully be able to avoid the code above.